In [1]:
import pandas as pd
import scipy.stats as stats

df = pd.read_csv("/content/teachers_rating_data (3).csv")

print(df.head())

     prof  Gender  Age Tenure  Evaluation  Students  Beauty CourseLevel
0  Prof38    Male   33    Yes        4.87        59     4.8       Lower
1  Prof38    Male   43    Yes        4.39        97     2.8       Lower
2  Prof46  Female   29     No        3.96        96     3.2       Lower
3  Prof46    Male   43     No        4.55        43     3.8       Lower
4  Prof13  Female   43    Yes        4.94        96     3.1       Lower


# Q1. T-Test: Using the teachers' rating data set, does gender affect teaching evaluation rates?

In [2]:
correlation, p_value = stats.pearsonr(df['Beauty'], df['Evaluation'])

print(f"Correlation between Beauty and Evaluation: {correlation:.4f}")
print(f"P-value: {p_value:.4f}")

Correlation between Beauty and Evaluation: 0.0417
P-value: 0.6509


Correlation Coefficient: 0.0417

This value indicates a very weak positive linear relationship between 'Beauty' and 'Evaluation'.
In practical terms, this means that as a professor's perceived beauty score increases, their evaluation score tends to slightly increase — but the effect is extremely small and negligible.

P-value: 0.6509

Since this p-value is greater than the standard significance level of 0.05, the correlation is not statistically significant.
This means we fail to reject the null hypothesis, indicating that the weak positive relationship observed in the sample could easily be due to random variation and does not provide strong evidence of a real relationship in the overall population.


# Q2. ANOVA: Using the teachers' rating data set, does beauty score for instructors differ by age?

## Objective:
To check whether beauty scores differ significantly across different age groups.

## Hypotheses:

H₀: Mean beauty scores are equal across all age groups.

H₁: At least one group has a different mean beauty score.

## Test Used:
One-way ANOVA

In [3]:
male_eval = df[df['Gender'] == 'Male']['Evaluation']
female_eval = df[df['Gender'] == 'Female']['Evaluation']

t_stat, p_value_gender = stats.ttest_ind(male_eval, female_eval)

print(f"T-statistic: {t_stat:.4f}")
print(f"P-value: {p_value_gender:.4f}")

T-statistic: 2.0053
P-value: 0.0472


T-statistic: 2.0053

This positive value indicates that the average 'Evaluation' score for male professors is slightly higher than the average score for female professors.
However, the difference in means is relatively small.

P-value: 0.0472

Since this p-value is slightly less than the standard significance level of 0.05, the result is statistically significant.
We therefore reject the null hypothesis, indicating that there is a significant difference in the mean evaluation scores between male and female professors in this dataset.


# Q3. Chi-square: Using the teachers' rating data set, is there an association between tenure and gender?

# Objective:
To test whether tenure and gender are independent or associated.

# Hypotheses:

H₀: Tenure and gender are independent.

H₁: There is an association between tenure and gender.

Test Used:
Chi-square test of independence

In [4]:
contingency_table = pd.crosstab(df['Tenure'], df['Gender'])
chi2, p, dof, expected = stats.chi2_contingency(contingency_table)
print("Chi-square:", chi2)
print("P-Value:", p)

Chi-square: 0.08014503878474447
P-Value: 0.7771009549277145


# Interpretation:

fail to reject H₀ → They are independent.

# Q4. Correlation: Using the teachers' rating dataset, is teaching evaluation score correlated with beauty score?

# Objective:
To test whether there is a linear relationship between evaluation and beauty.

# Hypotheses:

H₀: There is no correlation between evaluation and beauty.

H₁: There is a correlation between evaluation and beauty.

Test Used:
Pearson Correlation Coefficient

In [5]:
tenured_eval = df[df['Tenure'] == 'Yes']['Evaluation']
non_tenured_eval = df[df['Tenure'] == 'No']['Evaluation']

t_stat_tenure, p_value_tenure = stats.ttest_ind(tenured_eval, non_tenured_eval)

print(f"T-statistic: {t_stat_tenure:.4f}")
print(f"P-value: {p_value_tenure:.4f}")

T-statistic: 0.6587
P-value: 0.5114


T-statistic: 0.6587  

This positive value indicates that the average 'Evaluation' score for tenured professors is slightly higher than that of non-tenured professors.  
However, the difference in means is very small and not practically meaningful.  

P-value: 0.5114  

Since this p-value is greater than the standard significance level of 0.05, the result is not statistically significant.  
Therefore, we fail to reject the null hypothesis, indicating that tenure status does not have a significant effect on teaching evaluation scores.
